plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -1705,6 +1705,11 @@ private void prepareDropTable(SchemaTableName schemaTableName)
                         try {
                             delegate.dropTable(schemaTableName.getSchemaName(), schemaTableName.getTableName(), true);
                         }
+                        catch (TableNotFoundException e) {
```

Comment:
This may suppress the exception even if dropTable method failed to delete the data, right? 

https://github.com/trinodb/trino/blob/a24475e3011c22cc107db0878d6b308cededb823/plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastore.java#L1062-L1090

Reply:
Yes. But even when exception is not suppressed, data will not be deleted neither, because next query fail as well.
Do you think we should improve data cleanup in scope of this PR?

Reply:
From a user perspective, doing a `DROP TABLE` may issue now a `TableNotFoundException` in the situation where the metastore takes too much time to delete the files of the managed table.
With the changes from @pajaks the statement `DROP TABLE` completes successfully even if it may be dealing with a `TableNotFoundException` situation, which (rather unlikely) may leave some table files behind.

Reply:
This is a corner case only when the metastore does not delete the files and user explicitly configure `hive.metastore.thrift.delete-files-on-drop`